### DISCUSSION TITLE
Using vue-language-server version 2.0

### DISCUSSION THREAD

**Discussion by hirotaka** [LazyVim/LazyVim#2814] [Q&A] [answered]
I'm installing and using vue-language-server through Mason for Vue development.

Since the server is now version 2, it seems to require a different setup.
Its README shows the following steps:
https://github.com/vuejs/language-tools?tab=readme-ov-file#hybrid-mode-configuration-requires-vuelanguage-server-version-200

I'm trying to set it up with LazyVim, but it is not working.
Has anyone been able to set this up?

**Comment by hirotaka** [comment, marked as answer]
It looks like it's working with the following config.

Crete `lua/plugins/vue.lua` file with:

```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      setup = {
        tsserver = function(_, opts)
          local mason_registry = require("mason-registry")
          local vue_language_server_path = mason_registry.get_package("vue-language-server"):get_install_path()
            .. "/node_modules/@vue/language-server"

          opts.init_options = {
            plugins = {
              {
                name = "@vue/typescript-plugin",
                location = vue_language_server_path,
                languages = { "vue" },
              },
            },
          }
          opts.filetypes = { "typescript", "javascript", "javascriptreact", "typescriptreact", "vue" }
        end,
      },
      servers = {
        tsserver = {},
      },
    },
  },
}

```

Is there any better way to do this?

**Reply by yunxi177** [reply]
I found that in .vue files, there seems to be an issue with object property suggestions. For example:
```
const form = reactive({
  id: 1,
});

```
When I type form., there is no suggestion for id, but it works fine in .ts files.
